### PR TITLE
Log CCR cluster connection

### PR DESCRIPTION
**Pull request by idegtiarenko**
Today we only log if the CCR connection timed out. This change logs successful connection as well as failure if any.

### CODE REVIEW

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
Actually no I take that back, sorry :)

This will emit logs on settings updates even if nothing changes about the remote cluster, which could be very noisy. We need to only emit these logs if we actually did something.

**Comment by idegtiarenko** [comment]
>This will emit logs on settings updates even if nothing changes about the remote cluster, which could be very noisy. We need to only emit these logs if we actually did something.

I just checked and it is the case. This code is triggered from:
https://github.com/elastic/elasticsearch/blob/5dc2541675b897f1d64ceec514f76c22a5c3bf63/server/src/main/java/org/elasticsearch/transport/RemoteClusterAware.java#L103-L116

that is only looking on a subset of all settings. While testing I also realized this will log `connected to new remote cluster` when setting seeds to null, so the log message should be updated. I think all this justifies a test case.

File: server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java [hunk 1]
```diff
@@ -309,13 +309,13 @@ protected void updateRemoteCluster(String clusterAlias, Settings settings) {
         CountDownLatch latch = new CountDownLatch(1);
         updateRemoteCluster(clusterAlias, settings, ActionListener.runAfter(new ActionListener<>() {
             @Override
-            public void onResponse(Void o) {
-                logger.debug("connected to new remote cluster [{}]", clusterAlias);
+            public void onResponse(RemoteClusterConnectionStatus status) {
+                logger.info("remote cluster connection [{}] updated: {}", clusterAlias, status);
```
Comment (fcofdez) [inline server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java:313]: I wonder if this could be too noisy in certain scenarios?

**Comment by fcofdez** [comment]
This looks good, but maybe we should add a test check that when a setting changes we don't log a connection change?

File: server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java (hunk 1 above)
Comment (idegtiarenko) [reply server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java:313]: To clarify this:
* it is only logged when updating the relevant setting, or other cluster settings are not triggering this message
* It is not logged when starting or stopping the node
* It is logged from EVERY node of the cluster. I think it is better to keep it this way rather then make it look like only elected master is trying to connect to remote.

**Review by DaveCTurner — APPROVED** [review]
LGTM this makes sense to me and it doesn't seem too noisy either.

**Review by fcofdez — APPROVED** [review]
LGTM. Sorry for the delay

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/bwc

**Pull request merged**